src/app/clusters/thread-network-directory-server/DefaultThreadNetworkDirectoryStorage.cpp

```diff
@@ -118,11 +118,18 @@ CHIP_ERROR DefaultThreadNetworkDirectoryStorage::RemoveNetwork(const ExtendedPan
     VerifyOrReturnError(FindNetwork(exPanId, index), CHIP_ERROR_NOT_FOUND);
 
     // Move subsequent elements down to fill the deleted slot
-    for (mCount--; index < mCount; index++)
+    index_t subsequent = --mCount - index;
```

Comment:
This might be more readable as:
```suggestion
    index_t subsequent_count = mCount - (index + 1);
    --mCount;
```